clients/postgres/idp.go

```diff
@@ -0,0 +1,37 @@
+package postgres
+
+import (
+	"database/sql"
+
+	"github.com/mainflux/mainflux/clients"
+	"github.com/mainflux/mainflux/logger"
+	uuid "github.com/satori/go.uuid"
+)
+
+var _ clients.IdentityProvider = (*identityProvider)(nil)
+
+type identityProvider struct {
```

Comment:
This is over-engineering. `Identity` check can be performed as part of the queries in postgres repository, while Key generation can be kept inside the repository (as before).